torch/jit/__init__.py

```diff
@@ -1083,8 +1107,69 @@ def define(self, lang):
             # we add 1 to get to the proper surrounding scope.
             rcb = createResolutionCallback(frames_up=1)
             self._define(lang, rcb, True)
+
+    class WeakScriptModuleProxy(ScriptModule):
+        def __init__(self, original):
+            methods = []
+            # Link to script module on original object
+            original._jit_script_module = self
+            self.__dict__['original'] = original
+
+            # Convert weak_methods to ScriptMethodStubs for compilation when
+            # __init__ is called
+            for item in dir(original):
+                func = getattr(original, item)
+                if not callable(func) or not isinstance(func, types.MethodType):
+                    continue
+                if func.__func__ in weak_script_methods:
+                    entry = weak_script_methods[func.__func__]
+                    stub = script_method(entry.original_method, entry.resolution_callback)
+                    methods.append(stub)
+                    del weak_script_methods[func.__func__]
+
+            super(WeakScriptModuleProxy, self).__init__(
+                _weak_copier=self._copy_module_data, _new_stubs=methods)
+
+            # Hook up forwarding getters and setters
+            getter = self._weak_getattr
+            setter = self._weak_setattr
+            self.__getattr__ = getter
+            self.__setattr__ = setter
+
+        def _weak_getattr(self, attr):
+            return getattr(self.original, attr)
+
+        def _weak_setattr(self, attr, value):
+            return setattr(self.original, attr, value)
+
+        def _copy_module_data(self):
+            original = self.__dict__['original']
+
+            # Copy constants
+            self.__dict__['__constants__'] = original.__constants__
```

Comment:
`original` is not a ScriptModule, so it will never has a `_constants_set`. It should be fine to just grab `__constants__` from original and assign it to self._constants_set.